### ISSUE TITLE
Style/NestedParenthesizedCalls shouldn't flag single argument methods

### ISSUE THREAD

**Issue by CyborgMaster** [bbatsov/rubocop#2705]
This cop wants me to replace:
``` ruby
Mailer.expects(:mail).with(has_entry subject: 'test subject').returns(stub_everything)
```
with:
``` ruby
Mailer.expects(:mail).with(has_entry(subject: 'test subject')).returns(stub_everything)
```

While I agree with the premise of this cop and feel that the example give is good:
``` ruby
method1(method2 arg, method3 arg) -> method1(method2(arg), method3(arg))
```
I don't think changing
``` ruby
method1(method2 arg) -> method1(method2(arg))
```
improves anything.  I think the extra parenthesis makes it harder to read.  I vote that when there is only one argument to the outer function, this inner parentheses should not be required.  

What do you think?

```
$ rubocop -V
0.36.0 (using Parser 2.3.0.1, running on ruby 2.2.2 x86_64-darwin14)
```

**Comment by jfelchner** [comment]
FWIW, I ran into this and tend to agree, although I'm not vehemently against adding the parens regardless.

**Comment by alexdowad** [comment]
> I don't think changing
> 
> method1(method2 arg) -> method1(method2(arg))
>
> improves anything. I think the extra parenthesis makes it harder to read. I vote that when there is only one argument to the outer function, this inner parentheses should not be required.

The original proposal to add this cop specifically pointed to a case where there was a single call nested inside the parens, and was accepted on that basis.

Perhaps you can simply disable the cop?

**Comment by jonas054** [comment]
Also, the [style guide](https://github.com/bbatsov/ruby-style-guide#no-dsl-parens) says

>  Omit parentheses around parameters for methods that are part of an internal DSL (e.g. Rake, Rails, RSpec), methods that have "keyword" status in Ruby (e.g. attr_reader, puts) and attribute access methods. Use parentheses around the arguments of all other method invocations.

So I think the cop's default behavior needs to remain as it is. Adding another style that allows leaving out the parentheses in certain situations is an option, but it adds complexity.

**Comment by alexdowad** [comment]
Any further comments on this question?

**Comment by CyborgMaster** [comment]
Based on the style guide I see that my instinct on this one was wrong.  The cop is behaving as it should.  I think I will just disable it for our codebase.  Thanks for taking the time to explain it to me.
